tests/pyanaconda_tests/screen_access_manager_test.py

```diff
@@ -0,0 +1,253 @@
+# vim:set fileencoding=utf-8
+#
+# Copyright (C) 2016  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Martin Kolman <mkolman@redhat.com>
+#
+
+import unittest
+import tempfile
+import shutil
+import os
+from configparser import ConfigParser
+
+from pyanaconda import screen_access
+
+class ScreenAccessManagerTest(unittest.TestCase):
+    """Test the spoke access manager.
+
+    Check that it can correctly handle existing user interaction config files,
+    correctly handle spoke visits and option changes, state querying and
+    that the config files it creates are valid.
+    """
+
+    def _get_config_path(self):
+        test_dir = tempfile.mkdtemp()
+        config_path = os.path.join(test_dir, screen_access.CONFIG_FILE_NAME)
+        return test_dir, config_path
+
+    def _parse_config_file(self, config_path):
+        config = ConfigParser()
+        with open(config_path, "rt") as f:
+            config.read_file(f)
+        return config
+
+    def file_header_test(self):
+        """Test that the Anaconda generated config file header looks fine."""
+        screen_access.initSAM()
+        sam = screen_access.sam
+        header = sam._get_new_config_header()
+        # check that the header starts with a hash
+        self.assertEqual(header[0], "#")
+        # check that it has a newline at the end
+        self.assertEqual(header[-1], "\n")
+
+    def no_interaction_test(self):
+        """Test that SAM can handle no user interaction taking place."""
+        screen_access.initSAM()
+        sam = screen_access.sam
+
+        try:
+            test_dir, config_path = self._get_config_path()
+            sam.write_out_config_file(os.path.join(test_dir, screen_access.CONFIG_FILE_NAME))
+            self.assertTrue(os.path.exists(config_path))
+            # the config should have the "created by Anaconda" header, so it should
+            # be non-zero
+            self.assertGreater(os.path.getsize(config_path), 0)
+        finally:
+            shutil.rmtree(test_dir)
```

Comment:
Is the try..finally a good idea? What if some of your functions raise an exception? Shouldn't it be propagated so the test will fail.